CRM/Volunteer/Form/VolunteerSignUp.php

```diff
@@ -232,38 +339,84 @@ function postProcess() {
     $values = $this->controller->exportValues();
 
     $profileFields = array();
-    foreach ($this->getProfileIDs() as $profileID) {
+    foreach ($this->getPrimaryVolunteerProfileIDs() as $profileID) {
       $profileFields += CRM_Core_BAO_UFGroup::getFields($profileID);
     }
     $profileValues = array_intersect_key($values, $profileFields);
     $activityValues = array_diff_key($values, $profileValues);
```

Comment:
Your IDE is correct, and I've adapted the function to take the two missing vars as params